tsfc/fem.py

```diff
@@ -108,6 +112,88 @@ def translator(self):
         return Translator(self)
 
 
+class CoordinateMapping(PhysicalGeometry):
+    """Callback class that provides physical geometry to FInAT elements.
+
+    Required for elements whose basis transformation requires physical
+    geometry such as Argyris and Hermite.
+
+    :arg mt: The modified terminal whose element will be tabulated.
+    :arg interface: The interface carrying information (generally a
+        :class:`PointSetContext`).
+    """
+    def __init__(self, mt, interface):
+        super().__init__()
+        self.mt = mt
```

Comment:
Chuck in the whole modified terminal, so we access to restrictions (for interior facet integrals) etc.